flask_project/campaign_manager/views.py

```diff
@@ -1039,48 +911,127 @@ def create_campaign():
+         )
+    form.start_date.data = campaign_obj.start_date
+    if campaign_obj.end_date:
+        form.end_date.data = campaign_obj.end_date
+    context['allowed_managers'] = managers
+    context['oauth_consumer_key'] = OAUTH_CONSUMER_KEY
+    context['oauth_secret'] = OAUTH_SECRET
+    context['map_provider'] = map_provider()
+    context['action'] = 'edit'
+    context['functions'] = get_selected_functions()
+    context['title'] = 'Edit Campaign'
+    context['maximum_area_size'] = MAX_AREA_SIZE
+    context['uuid'] = campaign_obj.uuid
+    context['types'] = {}
+    context['campaign_creator'] = campaign_obj.creator.osm_user_id
+    context['link_to_omk'] = campaign_obj.link_to_OpenMapKit
+    try:
+        context['types'] = json.dumps(
+            get_types()).replace('True', 'true').replace('False', 'false')
+    except ValueError:
+        pass
+    return (context, form)
+
+
+@campaign_manager.route('/copy/<uuid>', methods=['GET', 'POST'])
+def copy_campaign_data(uuid):
+    from campaign_manager.forms.campaign import CampaignForm
+    import uuid as _uuid
+    from flask import url_for, redirect
+    campaign = Campaign()
+    campaign_obj = campaign.get_by_uuid(uuid)
+    if request.method == 'GET':
+        context, form = get_edit_copy_arguments(campaign_obj)
+
+    else:
+        form = CampaignForm(request.form)
+        if form.validate_on_submit():
+            data = form.data
+            data.pop('types_options')
+            data.pop('csrf_token')
+            data.pop('submit')
+            data['new_uuid'] = _uuid.uuid4().hex
+            user = User()
+            new_creator = user.get_by_osm_id(data['uploader'])
+            campaign_copy = Campaign(
+                name=data['name'],
+                description=data['description'],
+                creator_id=new_creator.id,
+                start_date=data['start_date'],
+                end_date=data['end_date'],
+                create_on=datetime.now(),
+                uuid=data['new_uuid'],
+                version=2,
+                map_type=data['map_type']
+            )
+            campaign_copy.create()
+            campaign_copy.save_feature_types(data)
+            campaign_copy.save_geometry(data)
+            campaign_copy.save_insight_functions(data)
+            campaign_copy.save_managers(data)
+            create_thumbnail_image(campaign_copy)
+            return redirect(
+                    url_for('campaign_manager.get_campaign',
+                            uuid=campaign_copy.uuid)
+                    )
+    context['url'] = '/edit/copy/%s' % uuid
+    context['mode'] = 'copy'
+    context['campaign'] = campaign_obj
+    try:
+        context['types'] = json.dumps(
+            get_types()).replace('True', 'true').replace('False', 'false')
+    except ValueError:
+        pass
+    return render_template(
+        'create_campaign.html', form=form, **context)
+
+
 @campaign_manager.route('/edit/<uuid>', methods=['GET', 'POST'])
```

Comment:
could we have one function per method? Better clarity and less complexity.
Following this [guide](https://restfulapi.net/resource-naming/), we could have those new routes:

GET /campaigns/<uuid> -> show a campaign
GET /campaigns/new -> show the campaign form
POST /campaigns -> create a campaign
GET /campaigns/<uuid>/edit -> show the campaign form with campaign data
PUT /campaigns/<uuid> -> update the edited campaign